proto/quotes/quote_service.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+package services.quotes.v1;
+
+service QuoteService {
+  rpc GetQuoteToBuyCents(GetQuoteToBuyCentsRequest) returns (GetQuoteToBuyCentsResponse) {}
+  rpc GetQuoteToSellCents(GetQuoteToSellCentsRequest) returns (GetQuoteToSellCentsResponse) {}
+  rpc AcceptQuote(AcceptQuoteRequest) returns (AcceptQuoteResponse) {}
+}
+
+message GetQuoteToBuyCentsRequest {
+  oneof quote_for {
+    uint64 amount_to_sell_in_sats = 1;
```

Comment:
one might want to "sell $100 worth of bitcoin", so the amount as an input should be dollar, not sats.

Reply:
That’s what the next line is for. You can specify either the btc amount to sell or the USD amount to buy (your scenario). See the oneof keyword

Reply:
it's a different use case. yours is `buy with a USD amount`. the one I mention i `sell with a USD amount`

Reply:
There is a matrix of 2*2*2 = 8 use cases.

The 3 binary variables are:
buy / sell USD
denominate amount in BTC / USB
execute immediately / execute in future

The api of the current price server has each of the 8 use cases as its own function call.
In this api I'm proposing to reduce the number of exposed endpoint to 2 (buy / sell) and pass the other binaries as function variables (denomination / execution).

Reply:
If as a user, I request to sell $100 of bitcoin, but get credited 99.98 because of some math approximation and because if a mismatch of source of truth, then I would think there are hidden fees in here or that the system doesn't really work well.

If stablesats is going to be the source of truth, we can make those shortcuts because it would worsen the user experience and reduce trust in the system

Reply:
So in your example the request would call `GetQuoteToBuyCents(GetQuoteToBuyCentsRequest)` because we are buying USD (ie. selling bitcoin). But we want to denominate the amount in `use` ($100 worth). That means in the `oneof` argument I would specify the `amount_to_buy_in_cents = 100`

```
message GetQuoteToBuyCentsRequest {
  oneof quote_for {
    uint64 amount_to_sell_in_sats = 1;
    uint64 amount_to_buy_in_cents = 2;
  }

  bool immediate_execution = 3;
}
=>
GetQUoteToBuyCentsRequest {
  quote_for: { amount_to_buy_in_cents: 100 }
  immediate_execution: false
}
```

This is the same way it works now with the [`GetSatsFromCentsForFutureBuy`](https://github.com/GaloyMoney/stablesats-rs/blob/469344f1c5faf2909858a35c017bf313f9461e45/proto/price/price_service.proto#L11) call.

The only difference here is that we are passing some arguments instead of separating each method call.

Reply:
ok, I misread the initial interface.

If I want to buy $100 of bitcoin, can I get a quote by passing a dollar amount?

Reply:
Yes (see example above)

Reply:
ok.

I think `message GetQuoteToBuyCentsRequest`

would be better worded with`message GetQuoteToBuyDollarRequest` or `message GetQuoteToBuyStablesatsRequest` 

because otherwise it's a bit convoluted that we using cents as both the assets and unit of account. 

This may be why I have been mis interpreting the API (I skimed through it and didnt' try to read it thoroughly)